helix-term/src/ui/mod.rs

```diff
@@ -339,6 +339,17 @@ pub mod completers {
         }
     }
 
+    pub fn language_servers(editor: &Editor, input: &str) -> Vec<Completion> {
+        let language_servers = doc!(editor)
+            .language_servers()
+            .map(|ls| ls.name().to_string());
+
+        fuzzy_match(input, language_servers, false)
+            .into_iter()
+            .map(|(name, _)| ((0..), Span::raw(name)))
```

Comment:
The `to_string()` can move from the `map` in `let language_servers` to the `map` here on L349. That way we allocate only for names that match the input.